graph/memstore/quadstore.go

```diff
@@ -117,17 +116,17 @@ func (qs *QuadStore) ApplyDeltas(deltas []graph.Delta, ignoreOpts graph.IgnoreOp
 		case graph.Add:
 			if !ignoreOpts.IgnoreDup {
 				if _, exists := qs.indexOf(d.Quad); exists {
-					return graph.ErrQuadExists
+					return &graph.QuadError{Action: d.Action, Quad: d.Quad, Err: graph.ErrQuadExists}
```

Comment:
Why? They are all errors which describe failures in delta application. This is almost entirely analogous to the use of os.PathError.

If we document that methods return a QuadError on failure then callers can assert if they want to find the error.